### PR TITLE
Binomial thm

### PR DESCRIPTION
**Pull request by mzeuner**
The aim of this pull request is to ultimately prove the binomial theorem for arbitrary commutative rings.
A lot of definitions and results about finite sums have been upstreamed from `Cubical.Algebra.Matrix`  to `Cubical.Algebra.Ring.BigOps`

### CODE REVIEW

File: Cubical/Algebra/Monoid/BigOp.agda [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  ≡⟨ cong (V zero ·_) (assoc _ _ _) ⟩
+    V zero · ((W zero · bigOp (V ∘ suc)) · bigOp (W ∘ suc))
+  ≡⟨ cong (λ x → V zero · (x · bigOp (W ∘ suc))) (comm _ _) ⟩
+    V zero · ((bigOp (V ∘ suc) · W zero) · bigOp (W ∘ suc))
+  ≡⟨ cong (V zero ·_) (sym (assoc _ _ _)) ⟩
+    V zero · (bigOp (V ∘ suc) · (W zero · bigOp (W ∘ suc)))
+  ≡⟨ assoc _ _ _ ⟩
+    V zero · bigOp (V ∘ suc) · (W zero · bigOp (W ∘ suc)) ∎
+
+ bigOpTop : (∀ x y → x · y ≡ y · x)
```
Comment (mortberg) [inline Cubical/Algebra/Monoid/BigOp.agda:71]: `bigOpLast` and put the last thing on the right instead

File: Cubical/Algebra/Ring/BigOps.agda [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+   V zero · x + (∑ (V ∘ suc)) · x       ≡⟨ (λ i → V zero · x + ∑Mulldist x (V ∘ suc) i) ⟩
+   V zero · x + ∑ (λ i → V (suc i) · x) ∎
+
+ ∑Mulr0 : ∀ {n} → (V : FinVec R n) → ∑ (λ i → V i · 0r) ≡ 0r
+ ∑Mulr0 V = sym (∑Mulldist 0r V) ∙ 0RightAnnihilates _
+
+ ∑Mul0r : ∀ {n} → (V : FinVec R n) → ∑ (λ i → 0r · V i) ≡ 0r
+ ∑Mul0r V = sym (∑Mulrdist 0r V) ∙ 0LeftAnnihilates _
+
+ ∑Mulr1 : (n : ℕ) (V : FinVec R n) → (j : Fin n) → ∑ (λ i → V i · (if i == j then 1r else 0r)) ≡ V j
```
Comment (mortberg) [inline Cubical/Algebra/Ring/BigOps.agda:75]: Diracify

Comment (mortberg) [reply Cubical/Algebra/Ring/BigOps.agda:75]: Oops, Kronecker!

File: Cubical/Data/Nat/Properties.agda [hunk 3]
```diff
@@ -74,6 +74,15 @@ suc-predℕ (suc n) p = refl
 +-suc zero    n = refl
 +-suc (suc m) n = cong suc (+-suc m n)
 
++-one : ∀ m → m + 1 ≡ suc m
++-one zero = refl
++-one (suc m) = cong suc (+-one m)
+
++-two : ∀ m → m + 2 ≡ suc (suc m)
++-two zero = refl
++-two (suc m) = cong suc (+-two m)
```
Comment (ecavallo) [inline Cubical/Data/Nat/Properties.agda:83]: Is the particular way these are proved important? If not, they are special cases of `+-comm`.

**Comment by mzeuner** [comment]
It seems like Travis doesn't want to check the latest commit. Does anyone have an idea why?

**Pull request merged**